languages/csharp/exercises/concept/user-defined-exceptions/.meta/Example.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+
+public class CalculationException : Exception
+{
+    public CalculationException(int operand1, int operand2, string message, Exception inner) : base(message, inner)
+    // TODO: complete the definition of the constructor
+    {
+        Operand1 = operand1;
+        Operand2 = operand2;
+    }
+
+    public int Operand1 { get; }
+    public int Operand2 { get; }
+}
+
+public class CalculatorTestHarness
+{
+    private Calculator calculator;
+
+    public CalculatorTestHarness(Calculator calculator)
+    {
+        this.calculator = calculator;
+    }
+
+    public string Multiply(int x, int y)
+    {
+        try
+        {
+            TestMultiplication(x, y);
+        }
+        catch (CalculationException cex) when (cex.Operand1 < 0)
+        {
+            return "Multiply failed for a negative value. " + cex.InnerException.Message;
+        }
+        catch (CalculationException cex)
+        {
+            return "Multiply failed for a positive value. " + cex.InnerException.Message;
+        }
+
+        return "Multiply succeeded";
+    }
+
+    public void TestMultiplication(int x, int y)
+    {
+        try
+        {
+            calculator.Multiply(x, y);
+        }
+        catch (OverflowException ofex)
+        {
+            throw new CalculationException(x, y, string.Empty, ofex);
```

Comment:
I don't often see custom exceptions that don't have a message. Maybe we could have one here too like "An overflow occured while doing multiplication" or something like that?

Reply:
D) I changed this to string.Empty as we don't do anything with the message.  The inquisitive student might start wondering what the point of including a message was if it's never used.  Is there some magic behind the scenes logging that I don't see.

Reply:
Okay, agreed. Let's keep it like this. Maybe we can mention something on messages in exceptions in the `after.md` document.